tests/commands/new/test_build_gui_context.py

```diff
@@ -86,10 +86,11 @@ def main():
         pyproject_table_linux="""\
 requires = [
     "toga-gtk~=0.5.0",
-    # PyGObject 3.52.1 enforces a requirement on libgirepository-2.0-dev. This library
+    # PyGObject 3.50.0 is the last version that is known to be dependent on
+    # libgirepository1.0-dev. Newer releases depend on libgirepository-2.0-dev, which
     # isn't available on Debian 12/Ubuntu 22.04. If you don't need to support those (or
     # older) releases, you can remove this version pin. See beeware/toga#3143.
-    "pygobject < 3.52.1",
+    "pygobject <= 3.50.0",
```

Comment:
Going back to [this comment](https://github.com/beeware/briefcase/pull/2187#discussion_r1986461528), I think it may be better to pin this to only `3.50.0`. Versions less than `3.50.0` are not compatible with Toga.

Reply:
Fair enough. Toga's version pin will combine with this <= pin and only resolve to 3.50.0, so we might as well make the math easier for all concerned.